File: chapter1(1-50)/tenth/summation_prime.cpp
#include<iostream>
using namespace std;
int is_prime(long long int n){
   long long int i;
  if(n<=1)
    return 0;
  else
    for(i=2;i*i<=n;i++)
      if(n%i==0)
        return 0;
  return 1;
}
int main(){
  long long int i;
  long long int result=0;
  for(i=2;i< 2000000;i++)
    if(is_prime(i))
      result+=i;
  cout<<result<<endl;
  return 0;
}

File: chapter1(1-50)/tenth/summation_prime.rs
fn is_prime(n:i64)->bool {
    if n <= 1 {
        return false;
    }
    for i in 2..=(n as f64).sqrt() as i64 { //开根号要转化为f64
        if n % i == 0 {
            return false;
        }
    }
    true
}

fn  main(){
    let mut result:i64=0;
    for i in 2..2000000{
        if is_prime(i)==true{
            result+=i;
        }
    }
    println!("{}",result);
}
